pkg/config/config.go

```diff
@@ -0,0 +1,139 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	configapi "sigs.k8s.io/jobset/api/config/v1alpha1"
+)
+
+func fromFile(path string, scheme *runtime.Scheme, cfg *configapi.Configuration) error {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	codecs := serializer.NewCodecFactory(scheme, serializer.EnableStrict)
+
+	// Regardless of if the bytes are of any external version,
+	// it will be read successfully and converted into the internal version
+	return runtime.DecodeInto(codecs.UniversalDecoder(), content, cfg)
```

Comment:
Will this invoke SetDefaults_Configuration to populate the default values of the fields that are not set.

Reply:
Yes, it will call defaulter before assigning value.